test_data/rules/unused_top_members_in_executable_libraries.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `dart test -N unused_top_members_in_executable_libraries`
+
+import 'package:meta/meta.dart';
+
+main() // OK
+{
+  _f5();
+  f1();
+  f3(() {
+    f4(b);
+  });
+  f4(b);
+}
+
+const a = 1; // LINT
+const b = 1; // OK
+
+int v = 1; // LINT
+
+typedef A = String; // LINT
+
+class C {} // LINT
```

Comment:
Could you add two constructors here to show that the presence of constructors (which refer to the name of the class) does not count the class as "used"

```
class C { // LINT
  C();
  C.named();
}
```

Reply:
This comment still stands. I want to make sure that `traverseNodeInDFS` and your algorithm do not count `C.named();` as a legitimate reachability reference to `C`. `C` should be marked unreachable.

Reply:
done